cube/src/main/java/zone/cogni/asquare/cube/index/IndexingServiceContext.java

```diff
@@ -24,4 +31,31 @@ abstract class IndexingServiceContext {
 
   protected abstract Map<String, String> getQueryTemplateParameters();
 
+  protected ElasticsearchMetadataService getElasticsearchMetadataService() {
+    return elasticsearchMetadataService;
+  }
+
+  @Override
+  public void ensureIndexExists(@Nonnull String index) {
```

Comment:
ok, I got it. I use `InternalIndexingServiceUtils` because I do not like to put too much code in superclasses. This often becomes a pain if there are too many exceptions in subclasses.

Hence my design where I kept the classes for the most part separate and just delegate to the (badly named) Utils class